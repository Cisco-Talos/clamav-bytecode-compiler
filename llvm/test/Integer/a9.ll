; RUN: llvm-as %s -o - | llvm-dis > %t.ll
; RUN: diff %t.ll %s.out

; test 9 bits
;
@b = constant i9 add(i9 511, i9 1)
@c = constant i9 add(i9 511, i9 511)
@d = constant i9 add(i9 504, i9 8)
@e = constant i9 sub(i9 0 , i9 1)
@f = constant i9 sub(i9 0 , i9 511)
@g = constant i9 sub(i9 2 , i9 511)

@h = constant i9 shl(i9 1 , i9 9)
@i = constant i9 shl(i9 1 , i9 8)
@j = constant i9 lshr(i9 511 , i9 8)
@l = constant i9 ashr(i9 511 , i9 8)

@n = constant i9 mul(i9 511, i9 2)
@q = constant i9 sdiv(i9 511, i9 2)
@r = constant i9 udiv(i9 511, i9 2)
@s = constant i9 urem(i9 511, i9 -1)
@t = constant i9 srem(i9 1, i9 510)
@o = constant i9 trunc( i10 512  to i9 )
@p = constant i9 trunc( i10 511  to i9 )

