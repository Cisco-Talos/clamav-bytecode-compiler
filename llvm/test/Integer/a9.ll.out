; ModuleID = '<stdin>'

@b = constant i9 0                                ; <i9*> [#uses=0]
@c = constant i9 -2                               ; <i9*> [#uses=0]
@d = constant i9 0                                ; <i9*> [#uses=0]
@e = constant i9 -1                               ; <i9*> [#uses=0]
@f = constant i9 1                                ; <i9*> [#uses=0]
@g = constant i9 3                                ; <i9*> [#uses=0]
@h = constant i9 undef                            ; <i9*> [#uses=0]
@i = constant i9 -256                             ; <i9*> [#uses=0]
@j = constant i9 1                                ; <i9*> [#uses=0]
@l = constant i9 -1                               ; <i9*> [#uses=0]
@n = constant i9 -2                               ; <i9*> [#uses=0]
@q = constant i9 0                                ; <i9*> [#uses=0]
@r = constant i9 255                              ; <i9*> [#uses=0]
@s = constant i9 0                                ; <i9*> [#uses=0]
@t = constant i9 1                                ; <i9*> [#uses=0]
@o = constant i9 0                                ; <i9*> [#uses=0]
@p = constant i9 -1                               ; <i9*> [#uses=0]
